### PR TITLE
basic Asyncio support

### PR DESCRIPTION
**Pull request by jakkdl**
first steps on #208 
106, 200, 21x and 9xx are supported, though no autofix.

This is rebased on top of #209, so if reviewing you should only look at the latest commit(s).

todo:
* [x] update readme.md
* [x] update changelog
* [x] do a second pass through the diff, quite tired atm so suspect there may be silly stuff somewhere.
* [ ] rewrite descriptions of --anyio / --asyncio to not be dumb. And/or have a `--library` flag instead.

### CODE REVIEW

**Comment by Zac-HD** [comment]
Looks good so far!  Will re-review when you think it's done 👍

File: tests/test_flake8_async.py [hunk 1]
```diff
@@ -195,10 +188,16 @@ def check_autofix(
...
 @dataclass
 class MagicMarkers:
+    # Exclude checking a library against a file
     NOANYIO: bool = False
     NOTRIO: bool = False
+    NOASYNCIO: bool = False
+    # File should raise no errors with this library
     ANYIO_NO_ERROR: bool = False
-    ASYNC_NO_ERROR: bool = False
+    TRIO_NO_ERROR: bool = False
```
Comment (jakkdl) [inline tests/test_flake8_async.py:197]: this is a revert of an overeager search+replace in previous PR.

File: flake8_async/visitors/visitor_utility.py [hunk 2]
```diff
@@ -117,11 +117,13 @@ def __init__(self, *args: Any, **kwargs: Any):
         # see imports
         if self.options.anyio:
             self.add_library("anyio")
+        if self.options.asyncio:
+            self.add_library("anyio")
```
Comment (Zac-HD) [inline flake8_async/visitors/visitor_utility.py:121]:
```suggestion
        if self.options.asyncio:
            self.add_library("asyncio")
```

**Review by Zac-HD — COMMENTED** [review]
let's rebase on master, add `ASYNC102` support, and then we're probably done?

File: flake8_async/visitors/visitor_utility.py (hunk 2 above)
Comment (jakkdl) [reply flake8_async/visitors/visitor_utility.py:121]: huh, surprised this was not caught by a linter for "different branches of if with same content should be combined with `or`"

Comment (jakkdl) [reply flake8_async/visitors/visitor_utility.py]: oh derp, it's because they're separate `if`s - not different branches for the same `if`.

**Comment by jakkdl** [comment]
Uh, I ended up doing A Lot:tm:. But the diff didn't actually end up being all that big. The rebase might break viewing only new changes though, so you might want to review commit-by-commit.

Only ones listed in #208 not supported now is:
* 22X/23X/24X not having tailored error messages (and therefore being untested)
* not detecting asyncio-shielded-timeouts

So I'm now comfortable saying this has decent `asyncio` support and we could push a release.

TODO: Oh I did rename the repository, so should update links.

File: tests/test_config_and_args.py [hunk 3]
```diff
@@ -174,13 +174,24 @@ def test_anyio_from_config(tmp_path: Path, capsys: pytest.CaptureFixture[str]):
...
     )
-    err_file = str(Path(__file__).parent / "eval_files" / "anyio_trio.py")
-    expected = f"{err_file}:10:5: ASYNC220 {err_msg}\n"
+    err_file = Path(__file__).parent / "eval_files" / "anyio_trio.py"
+
+    # find the line with the expected error
+    for i, line in enumerate(err_file.read_text().split("\n")):
+        if "# ASYNC220: " in line:
+            # line numbers start at 1, enumerate starts at 0
+            lineno = i + 1
```
Comment (Zac-HD) [inline tests/test_config_and_args.py:183]:
```suggestion
    for lineno, line in enumerate(err_file.read_text().split("\n"), start=1):
        if "# ASYNC220: " in line:
```

https://docs.python.org/3/library/functions.html#enumerate

**Review by Zac-HD — APPROVED** [review]
LGTM; agree that we could release this and should release whatever we have on the weekend (when I set up release automation again).

I'll let you merge after any docs/links updates you want to make.

File: tests/test_config_and_args.py (hunk 3 above)
Comment (jakkdl) [reply tests/test_config_and_args.py:183]: :heart_eyes:

**Comment by jakkdl** [comment]
ah oops, I realized I forgot to ask your opinion on `use [trio|anyio].lowlevel.checkpoint()` vs `use [trio/anyio].lowlevel.checkpoint()` (or something else). Both are used in different places in documentation/comments/error messages

**Comment by Zac-HD** [comment]
I'd probably go with the slash, but don't really care so long as we're consistent.

**Pull request merged**